compiler/rustc_hir_analysis/src/check/check.rs

```diff
@@ -784,13 +784,15 @@ fn check_impl_items_against_trait<'tcx>(
             }
         }
 
-        check_specialization_validity(
-            tcx,
-            trait_def,
-            ty_trait_item,
-            impl_id.to_def_id(),
-            impl_item,
-        );
+        if !tcx.is_impl_trait_in_trait(ty_trait_item.def_id) {
```

Comment:
It does not make me feel comfortable to do fewer checks on RPITIT GATs. Especially because this PR has no comments anywhere.

This PR should be fixing the ICE that happens on the error reporting path.